R/zchunk_LA132.industry.R

```diff
@@ -48,48 +48,160 @@ module_energy_LA132.industry_DISABLED <- function(command, ...) {
+      filter(fuel == "gas") ->
+      L132.in_EJ_R_indunoil_F_Yh
+
+    ## Gas processing: This is complicated. Coal is not deducted, as inputs to coal gasification were mapped directly
+    # from the IEA energy balances, not derived from known fuel outputs multiplied by assumed IO coefs. The reason for doing this is that
+    # coal is one of several possible inputs to "gas works", and there is only one output. So no way to disaggregate fuel inputs if calculating from the output.
+    # Biogas is treated as primary energy in the IEA energy balances, so not relevant here.
+    ## Natural gas processing net energy use is the only one that may need to be calculated (09/2013: It is currently 0 as the IO coef is 1).
+    L122.in_EJ_R_gasproc_F_Yh %>%
+      filter(fuel == "gas") %>%
+      left_join_error_no_match(L122.in_EJ_R_gasproc_F_Yh, by = c("GCAM_region_ID", "sector", "fuel", "year")) %>%
+      left_join_error_no_match(L122.out_EJ_R_gasproc_F_Yh, by = c("GCAM_region_ID", "sector", "fuel", "year")) %>%
+      mutate(value = value.x - value.y) %>%
+      select(-value.x, -value.y) ->
+      L132.in_EJ_R_indgasproc_F_Yh
+
+    ## Refining: crude oil refining energy consumption is not derived based on the output and assumed IO coefs so it doesn't apply here
+    ## Refining: Electricity was taken off prior to scaling for end-use sectors, and in the IEA energy balances, biofuels are treated as primary energy
+    L122.in_EJ_R_refining_F_Yh %>%
+      filter(!grepl("oil refining", sector)) %>%
+      filter(fuel %in% c("gas", "coal")) ->
+      L132.in_EJ_R_indrefining_F_Yh
+
+    # CHP: no adjustments necessary
+    L123.in_EJ_R_indchp_F_Yh -> L132.in_EJ_R_indchp_F_Yh
+
+    # Combine all of the deduction tables and multiply by -1 to indicate that these are deductions
+    L132.in_EJ_R_indunoil_F_Yh %>%
+      bind_rows(L132.in_EJ_R_indrefining_F_Yh) %>%
+      bind_rows(L132.in_EJ_R_indgasproc_F_Yh) %>%
+      bind_rows(L132.in_EJ_R_indchp_F_Yh) %>%
+      mutate(value = -1 * value) ->
+      L132.in_EJ_R_inddeductions_F_Yh
+
+    ## Heat: fuel inputs to heat need to be added to industrial energy use, in regions where heat is not modeled as a final fuel
+    # Calculate the share of heat consumed by the industrial sector, in regions where heat is not modeled as a separate fuel
+    L131.share_R_Senduse_heat_Yh %>%
+      filter(grepl("industry", sector)) %>%
+      select(-sector) %>%
+      group_by(GCAM_region_ID, fuel, year) %>%
+      summarise(value = sum(value)) %>%
+      ungroup ->
+      L132.share_R_indenergy_heat_Yh
+
+    # Multiply these shares by the energy inputs to heat
+    A_regions %>%
+      filter(has_district_heat == 0) %>%
+      select(GCAM_region_ID) %>%
+      unlist ->
+      has_district_heat_GCAM_region_ID
+
+    L124.in_EJ_R_heat_F_Yh %>%
+      filter(GCAM_region_ID %in% has_district_heat_GCAM_region_ID) %>%
+      left_join(L132.share_R_indenergy_heat_Yh, by = c("GCAM_region_ID", "year")) %>%
```

Comment:
Why `left_join` here? Do we expect NAs?

Reply:
Replaced using `left_join_error_no_match`.